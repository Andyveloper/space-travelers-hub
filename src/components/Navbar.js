import * as React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/images/planet.png';

const Navbar = () => {
  const activeClassName = 'underline';
  return (
    <div>
      <div>
        <div className="Logo">
          <img src={logo} alt="planet" />
          <p>Space Travelers Hub</p>
        </div>
      </div>
      <nav>
        <ul className="">
          <li>
            <NavLink
              to="/rockets"
              className={({ isActive }) => (isActive ? activeClassName : undefined)}
            >
              Rockets
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/missions"
              className={({ isActive }) => (isActive ? activeClassName : undefined)}
            >
              Missions
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/profile"
              className={({ isActive }) => (isActive ? activeClassName : undefined)}
            >
              My Profiles
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
  );
};

export default Navbar;
