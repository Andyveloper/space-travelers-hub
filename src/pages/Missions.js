const Missions = () => (
  <div><h1>Hola Mision</h1></div>
);

export default Missions;
