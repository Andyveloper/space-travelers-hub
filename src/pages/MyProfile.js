import React from 'react';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const { rockets } = useSelector((state) => state.rockets);
  const { missions } = useSelector((state) => state.missions);
  const rocketList = Object.keys(rockets);
  const missionList = Object.keys(missions);
  const reservedRockets = [];
  const joinedMissions = [];

  if (rocketList.length) {
    rocketList.forEach((item) => {
      if (rockets[item].reserved) {
        reservedRockets.push({
          id: reservedRockets.length + 1,
          name: rockets[item].name,
        });
      }
    });
  }

  if (missionList.length) {
    missionList.forEach((item) => {
      if (missions[item].joined) {
        joinedMissions.push({
          id: joinedMissions.length + 1,
          name: missions[item].name,
        });
      }
    });

    return (
      <div className="mycontainer">
        <div className="mycontainer-section">
          <h2 className="mycontainer-section__title">
            My Missions
          </h2>
          <ul className="mycontainer-section__box">
            {
              joinedMissions.length ? joinedMissions.map((item) => (
                <li key={item.id} className="element-name">{item.name}</li>
              )) : <li className="element-name">No Missions Ongoing</li>
            }
          </ul>
        </div>
        <div className="mycontainer-section">
          <h2 className="mycontainer-section__title">
            My Rockets
          </h2>
          <ul className="mycontainer-section__box">
            {
              reservedRockets.length ? reservedRockets.map((item) => (
                <li key={item.id} className="element-name">{item.name}</li>
              )) : <li className="element-name">No rockets selected</li>
            }
          </ul>
        </div>
      </div>
    );
  }
  return false;
};

export default MyProfile;
