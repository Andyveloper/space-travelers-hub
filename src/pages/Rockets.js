import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from '../components/Rocket';

const Rockets = () => {
  const { rockets } = useSelector((state) => state.rockets);
  const rocketList = Object.keys(rockets);

  return (
    <ul style={{ padding: '2rem' }}>
      {
        rocketList ? rocketList.map((rocket) => (
          <Rocket
            name={rockets[rocket].name}
            description={rockets[rocket].description}
            image={rockets[rocket].image}
            reserved={rockets[rocket].reserved}
            id={rockets[rocket].id}
            key={rockets[rocket].id}
          />
        )) : <li>For the moment there are no rockets available</li>
      }
    </ul>
  );
};

export default Rockets;
