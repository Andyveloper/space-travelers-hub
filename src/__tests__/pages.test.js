import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Rockets from '../pages/Rockets';
import MyProfile from '../pages/MyProfile';
import Missions from '../pages/Missions';
import Navbar from '../components/Navbar';

describe('Renders page Components', () => {
  it('Renders Rockets component', () => {
    const tree = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders My Profile component', () => {
    const tree = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders Navbar component', () => {
    const tree = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders Missions component', () => {
    const tree = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
