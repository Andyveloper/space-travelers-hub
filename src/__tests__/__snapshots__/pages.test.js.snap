// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders page Components Renders Missions component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <table>
          <thead>
            <tr>
              <th>
                Mission
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
              <th>
                 
              </th>
            </tr>
          </thead>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <table>
        <thead>
          <tr>
            <th>
              Mission
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
            <th>
               
            </th>
          </tr>
        </thead>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders page Components Renders My Profile component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mycontainer"
      >
        <div
          class="mycontainer-section"
        >
          <h2
            class="mycontainer-section__title"
          >
            My Missions
          </h2>
          <ul
            class="mycontainer-section__box"
          >
            <li
              class="element-name"
            >
              No Missions Ongoing
            </li>
          </ul>
        </div>
        <div
          class="mycontainer-section"
        >
          <h2
            class="mycontainer-section__title"
          >
            My Rockets
          </h2>
          <ul
            class="mycontainer-section__box"
          >
            <li
              class="element-name"
            >
              No rockets selected
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mycontainer"
    >
      <div
        class="mycontainer-section"
      >
        <h2
          class="mycontainer-section__title"
        >
          My Missions
        </h2>
        <ul
          class="mycontainer-section__box"
        >
          <li
            class="element-name"
          >
            No Missions Ongoing
          </li>
        </ul>
      </div>
      <div
        class="mycontainer-section"
      >
        <h2
          class="mycontainer-section__title"
        >
          My Rockets
        </h2>
        <ul
          class="mycontainer-section__box"
        >
          <li
            class="element-name"
          >
            No rockets selected
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders page Components Renders Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="nav"
      >
        <div
          class="logo"
        >
          <div
            class="logo-container"
          >
            <img
              alt="planet"
              class="logo-container__img"
              src="planet.png"
            />
            <p>
              Space Travelers Hub
            </p>
          </div>
        </div>
        <nav
          class="nav-container"
        >
          <ul
            class="nav-container__items"
          >
            <li>
              <a
                href="/rockets"
              >
                Rockets
              </a>
            </li>
            <li>
              <a
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li
              class="profile"
            >
              <a
                href="/profile"
              >
                My Profiles
              </a>
            </li>
          </ul>
        </nav>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="nav"
    >
      <div
        class="logo"
      >
        <div
          class="logo-container"
        >
          <img
            alt="planet"
            class="logo-container__img"
            src="planet.png"
          />
          <p>
            Space Travelers Hub
          </p>
        </div>
      </div>
      <nav
        class="nav-container"
      >
        <ul
          class="nav-container__items"
        >
          <li>
            <a
              href="/rockets"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li
            class="profile"
          >
            <a
              href="/profile"
            >
              My Profiles
            </a>
          </li>
        </ul>
      </nav>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders page Components Renders Rockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        style="padding: 2rem 4rem;"
      />
    </div>
  </body>,
  "container": <div>
    <ul
      style="padding: 2rem 4rem;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
